File: Data Structures & Algorithms/string-encode-and-decode/submission-16.py
class Solution:

    def encode(self, strs: List[str]) -> str:
        encode = ""
        for s in strs:
            encode += str(len(s)) + "$" + s  
        return encode

    def decode(self, s: str) -> List[str]:
        print(s)
        res = []
        i = 0
        n = len(s)
        while i < n:
            dL = ""
            while i < n and s[i] != "$":
                dL = dL + s[i]
                i += 1
            print(dL)                
            m = int(dL)
            j = 0
            decode = ""
            i = i + 1
            while i < n and j < m:
                decode += s[i]                
                j += 1
                i += 1
            res.append(decode)
            
        
        return res
            

            

File: Data Structures & Algorithms/anagram-groups/submission-16.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map<string,vector<string>> group;
        vector<vector<string>> ans ;
        for(auto &word : strs){
            vector<int> letters(26);

            for(int i = 0 ; i < word.size() ; i++){
                letters[word[i] - 'a'] += 1;
            }
            string key = "";
            for(int i = 0 ; i < 26 ; i++){
                key += to_string(letters[i]) + ",";
            }
            group[key].push_back(word);
        }
        for(auto &entry : group){
            ans.push_back(entry.second);
        }
        return ans;
    }
};
